rs/ledger_suite/icp/src/lib.rs

```diff
@@ -806,6 +806,26 @@ impl NotifyCanisterArgs {
 }
 
 /// Arguments taken by the account_balance candid endpoint.
+#[derive(Clone, Eq, PartialEq, Hash, Debug, CandidType, Deserialize, Serialize)]
+pub struct AccountIdentifierByteBuf {
+    pub account: ByteBuf,
+}
```

Comment:
I created a new struct for parsing the candid input arguments rather than modifying the type of the `account` field in the existing `BinaryAccountBalanceArgs` below to the type `ByteBuf`. With the latter approach, in all other cases (mainly tests, but also a few production calls from the governance canister), when converting the `AccountIdBlob` of type `ByteBuf` into an `AccountIdentifier`, we would need to (re-check) the length of the `ByteBuf`, whereas in the approach with a new `AccountIdentifierByteBuf`, we can perform this check once only for calls to the `account_balance` candid endpoint. I'd be happy to consider other better approaches though!